tensorpipe/test/transport/shm/connection_test.cc

```diff
@@ -107,57 +94,44 @@ TEST_P(ShmTransportTest, QueueWrites) {
   constexpr int numMsg = 2;
   constexpr size_t numBytes = (3 * kBufferSize) / 4;
   std::array<char, numBytes> garbage;
-  std::promise<void> writeScheduledProm;
-  std::promise<void> writeCompletedProm;
-  std::promise<void> readCompletedProm;
-  std::future<void> writeCompletedFuture = writeCompletedProm.get_future();
-  std::future<void> readCompletedFuture = readCompletedProm.get_future();
 
-  this->testConnection(
+  testConnection(
       [&](std::shared_ptr<Connection> conn) {
-        writeScheduledProm.get_future().get();
```

Comment:
This `writeScheduledProm` has been removed but hasn't been replaced with anything. Based on the test's name, the reason it was there was to ensure the writes had all been scheduled and queued before starting to do the reads.

Reply:
The purpose of the test (making sure writes are properly queued) is ensured implicitly by having payloads larger than half of the ringbuffer size, but I'll add some message passing to make it explicit.